decko/lib/decko/application.rb

```diff
@@ -1,90 +1,69 @@
-        require environment
-      end
-    end
 
+  class Application < Cardio::Application
     class << self
       def inherited base
-        super
         Rails.app_class = base
+
         add_lib_to_load_path!(find_root(base.called_from))
-        ActiveSupport.run_load_hooks(:before_configuration, base.instance)
+        ActiveSupport.run_load_hooks(:before_load_environment_config, base.instance)
       end
     end
 
-    def add_path paths, path, options={}
-      root = options.delete(:root) || Decko.gem_root
-      options[:with] = File.join(root, (options[:with] || path))
-      paths.add path, options
+    initializer :load_decko_environment_config,
+                before: :bootstrap, group: :all do
+      Cardio.add_path "lib/decko/config/environments", glob: "#{Rails.env}.rb"
+      paths["lib/decko/config/environments"].existent.each do |environment|
+        require environment
+      end
     end
 
-    def config
-      @config ||= begin
-        config = super
-
-        Cardio.set_config config
-
-        # any config settings below:
-        # (a) do not apply to Card used outside of a Decko context
-        # (b) cannot be overridden in a deck's application.rb, but
-        # (c) CAN be overridden in an environment file
+    initializer :deck_autoload, before: :set_autoload_paths do |app|
+      Cardio.set_config
+      ActiveSupport::Dependencies.autoload_paths += config.autoload_paths
 
-        # therefore, in general, they should be restricted to settings that
-        # (1) are specific to the web environment, and
-        # (2) should not be overridden
-        # ..and we should address (c) above!
+      # any config settings below:
+      # (a) do not apply to Card used outside of a Decko context
+      # (b) cannot be overridden in a deck's application.rb, but
+      # (c) CAN be overridden in an environment file
 
-        # general card settings (overridable and not) should be in cardio.rb
-        # overridable decko-specific settings don't have a place yet
-        # but should probably follow the cardio pattern.
+      # therefore, in general, they should be restricted to settings that
+      # (1) are specific to the web environment, and
+      # (2) should not be overridden
+      # ..and we should address (c) above!
 
-        # config.load_defaults "6.0"
-        config.autoloader = :zeitwerk
-        config.load_default = "6.0"
-        config.i18n.enforce_available_locales = true
-        # config.active_record.raise_in_transactional_callbacks = true
+      # general card settings (overridable and not) should be in cardio.rb
+      # overridable decko-specific settings don't have a place yet
+      # but should probably follow the cardio pattern.
 
-        config.allow_concurrency = false
-        config.assets.enabled = false
-        config.assets.version = "1.0"
+      config.load_defaults "6.0"    # note this isn't a setter method
```

Comment:
this could have a lot of weird effects, no?

Reply:
Thanks for finding what was breaking with this. I rolled it back, got stuff to work and now just re-added with that exception overriden. I think this is good if it doesn't break stuff, yes?